server/emails/index.js

```diff
@@ -0,0 +1,195 @@
+      }
+    }
+  }
+};
+
+/**
+ * Loads metadata about an email template file and compiles it as a Handlebars
+ * template. The metadata is derived from the file path and its YAML
+ * frontmatter. The Handlebars template is compiled from the file's contents
+ * (without the frontmatter).
+ *
+ * @param {string} file - The path to the file to load.
+ * @returns {EmailData} Metadata and a compiled Handlebars template.
+ */
+async function loadEmailData(file) {
+
+  const { contents, frontmatter } = await loadFileWithFrontmatter(file);
+
+  if (!frontmatter.subject) {
+    throw new Error(`Email template ${file} has no "subject" in its YAML frontmatter`);
+  } else if (!isString(frontmatter.subject)) {
+    throw new Error(`Email template ${file} has a "subject" that is not a string`);
+  }
+
+  const name = getBasename(getDirname(file));
+  const extension = getExtension(file);
+  const locale = getBasename(file, extension);
+  const format = emailExtensionToFormat(extension);
+
+  const template = compileHandlebars(contents);
+  logger.trace(`Email template "${name}" (locale: ${locale}, format: ${format}) compiled from ${getRelativePath(config.root, file)}`);
+
+  return {
+    format,
+    name,
+    locale,
+    subject: frontmatter.subject,
+    template
+  };
+}
+
+/**
+ * Return the Nodemailer format name for an email template file based on its
+ * extension:
+ *
+ * * `.txt` becomes `text`.
+ * * `.html` becomes `html`.
+ *
+ * The format names should correspond to Nodemailer email configuration
+ * properties (see {@link https://nodemailer.com/message/}).
+ *
+ * @param {string} extension - The extension of the email template file (with the dot).
+ * @returns {string} The corresponding format name.
+ */
+function emailExtensionToFormat(extension) {
+  switch (extension) {
+    case '.txt':
+      return 'text';
+    case '.html':
+      return 'html';
+    default:
+      throw new Error(`Unknown email template extension "${extension}"`);
+  }
+}
+
+/**
+ * Metadata of one of the email templates in the `server/emails` directory.
+ *
+ * @typedef {Object} EmailData
+ * @property {string} format - The email format (see {@link module:server/emails~emailExtensionToFormat}).
+ * @property {string} name - A name identifying the email (derived from the file's directory name).
+ * @property {string} locale - The email locale (derived from the filename without the extension).
+ * @property {string} [subject] - The subject of the email (extracted from the file's YAML frontmatter if present).
+ * @property {Function} template - A Handlebars template function compiled from the file's contents (without its YAML frontmatter).
+ */
+
+/**
+ * Nodemailer options for an email's contents.
+ *
+ * @typedef {Object} EmailOptions
+ * @property {string} subject - The email subject.
+ * @property {string} [html] - The HTML version of the email (if available).
+ * @property {string} text - The plain text version of the email.
+ */
```

Comment:
Comment blocks with the [@typedef](https://devdocs.io/jsdoc/tags-typedef) annotation are [JSDoc](https://devdocs.io/jsdoc/) type definitions. They document complex types (in this case objects with various properties) so that they can be used in other comment blocks, in this case [here](https://github.com/MediaComem/biopocket-backend/blob/a864b60e782a654489e12df53a71864ff8d8afef/server/emails/index.js#L26) and [here](https://github.com/MediaComem/biopocket-backend/blob/a864b60e782a654489e12df53a71864ff8d8afef/server/emails/index.js#L124).

This allows you to document, as in this example, the exact structure of the object you are returning instead of simply saying `@returns {Object} ...´, which would not be as helpful.